app.go

```diff
@@ -21,7 +27,28 @@
 	a.ctx = ctx
 }
 
-// Greet returns a greeting for the given name
-func (a *App) Greet(name string) string {
-	return fmt.Sprintf("Hello %s, It's show time!", name)
+func (a *App) Greet(url string) string {
+
+	_, err := a.GithubService.ParsePullRequestURL(url)
+
+	if err != nil {
+
+		return fmt.Errorf("invalid GitHub pull request url. error: %s", err).Error()
+	}
+
+	pullRequest, err := a.GithubService.GetPullRequest(context.TODO(), url)
+
+	if err != nil {
+
+		return fmt.Errorf("error getting pull request. error: %s", err).Error()
+	}
+
+	var combinedDiff = ""
+
+	for _, diff := range pullRequest {
+
+		combinedDiff += fmt.Sprintf("%s\n%s\n---\n", diff.FilePathNew, diff.DiffContents)
+	}
+
+	return combinedDiff
```

Comment:
Also valid, but skipped for sake of rapid prototyping.  Since I will be doing some janitorial work here shortly, I will include this change in the next commit.

Reply:
Please make note that the use of the error, yet returning it as a string, is to signal to the developer this is an error condition and we're not JUST sending a message back to the front end. 

(Remember, this function is invoked from a binding in typescript from the react application)

Reply:
I have amended the previous commit with these changes.